cocotb/share/lib/vhpi/VhpiCbHdl.cpp

```diff
@@ -550,28 +553,40 @@ int VhpiSignalObjHdl::set_signal_value(long value)
         case vhpiLogicVecVal: {
             int i;
             for (i=0; i<m_num_elems; i++)
-                m_value.value.enumvs[m_num_elems-i-1] = value&(1L<<i);
-
-            // Since we may not get the numElems correctly from the sim and have to infer it
-            // we also need to set it here as well each time.
```

Comment:
Any reason for removing this comment? Still looks relevant to me.

Reply:
(it describes the assignment to .numElems)

Reply:
I just copy-pasted the other implementation over it. I'll put it back in.